### ISSUE TITLE
headers with a standard name like `string.h` and `limits.h` should be renamed

### ISSUE THREAD

**Issue by m6w6** [awesomized/libmemcached#61]
* *Reported as:* [lp:1436744](https://bugs.launchpad.net/bugs/1436744)
 * *Reported by:* [lp:~grobbelaar](https://launchpad.net/~grobbelaar)
 * *Reported at:* 2015-03-26T08:52:21Z

Imported from Launchpad using lp2gh.

------------------------------------

$ find . -name string.h
./third_party/libmemcached/libmemcached-1.0/struct/string.h
./third_party/libmemcached/libhashkit-1.0/string.h
./third_party/libmemcached/libhashkit/string.h
./third_party/libmemcached/tests/string.h

"string.h" is a header from the C runtime library. Project headers should not use this name.

**Comment by m6w6** [comment]
* *Comment by:* [lp:~grobbelaar](https://launchpad.net/~grobbelaar):
 * *Created at:* 2015-03-26T08:53:31Z

-----
These headers should be renamed, preferably each to a distinct name.

**Comment by cmb69** [comment]
Due to this, Windows in-tree builds of PECL/memcached fail, because other extensions include these files:
````
C:\php-snap-build\dep-aux\vs16\x64\libmemcached\include\libhashkit-1.0\string.h(27): error C2054: expected '(' to follow 'HASHKIT_API' (compiling source file ext\bz2\bz2.c)
…
````

**Comment by m6w6** [comment]
Yeah, `libmemcached/include/libhashkit-1.0` and siblings must not be in the include paths

**Comment by cmb69** [comment]
And that is [even possible](https://github.com/php/web-rmtools/commit/f660cbf5622e501c4fd1854885bab969a061098f). Thanks!
